python/arcticdb/version_store/library.py

```diff
@@ -1285,6 +1285,8 @@ def snapshot(
         ------
         InternalException
             If a snapshot already exists with ``snapshot_name``. You must explicitly delete the pre-existing snapshot.
+        MissingDataException
+            If ``allow_partial_snapshot`` is ``False`` and any ``versions`` specified cannot be found in the library.
```

Comment:
Users of this API can't provide the `allow_partial_snapshot` argument, you'll need to explain the circumstances of when this exception is thrown explicitly here.